### DISCUSSION TITLE
Align in c

### DISCUSSION THREAD

**Discussion by carlesoriol** [cc65/cc65#1513] [Q&A]
Is there any way to do the .align equivalent in C ?

**Comment by mrdudz** [comment]
not in C as such. some compilers (like GCC) implement attributes for that. With cc65 you can create a linker configuration that will align a certain segment as needed, and then you can put the data you want to align there.

**Reply by carlesoriol** [reply]
Yes I know not in C my question was about cc65. About linker configuration I can create a segment. But can I force this assignment before some data block (with pragma or similar).

**Comment by jmr** [comment]
C11 has _Alignas. It seems like adding support would be a reasonable feature request.

https://en.cppreference.com/w/c/language/_Alignas

**Comment by mrdudz** [comment]
It would be nice to have - but, unfortunately, due to .align works in ca65, this will not work well in most setups :/

**Comment by polluks** [comment]
Of course you can allocate a larger buffer and crop the margins with "&".
